### ISSUE TITLE
hiii <3

### ISSUE THREAD

**Issue by UnintelligentIdiot** [DriHut/ReFramed#5]
-%-First thank you for not ghosting this project, yet <33
Here's what I got for you.
Key-is used for blocks that have states(like stairs, fences, trapdoors and etc.) to lock their position like when used with debug stick, but its survival friendly. 
Screwdriver-Works on any theme block even if they don't have axis like planks( so we can have vertical planks <3). Just fun idea but it could be very useful, although that means that u will need to make it so it rotates on 6 sides every theme block...
Layered Panels-LP
1.It only returns 1 LP back when destroyed it doesn't matter if it has theme block in it or not.
2.When you have theme block in your left hand and LP in the right hand it uses the theme block on the left hand when trying to stack it, it shouldn't, and it only returns 1 of the theme blocks that has been used on the LP 
*I do love that the slabs and vertical slab(or panel) are in one block but I find it very hard to use it(And the other blocks that use this mechanic). Especially when I need to have a slab on the top of the block I have to be looking up to use it. Before I used Axioms "Angel Block" feature but it doesn't work anymore... and I haven't even started using in survival, I bet it would be really hard to use it
Maybe make the blocks have the same mechanic placement like the stairs do.

**Comment by DriHut** [comment]
Hello,
I'm glad you appreciate the work and very happy to get more feedback.

- So firstly the key I didn't notice there was such an item in FramedBlocks but I find the behavior very useful so I will add it in the v1.6 that I am currently working on.
- For the screwdriver there are three things to consider:
  1. The screwdriver only works on frames as letting it work on any block would be quite the horrible experience when working with other mods (its also the case for the key)
  2. Rotating planks and generally blocks textures is handled by the models in game so adding my own rotations on top of those would make everything more confusing when placing block that have different states. I do no exclude this feature as it would probably be useful in certain cases ( e.g. slopes which have a face that can have two different choice of texture because a slope has 5 faces against 6 in a cube)
  3. The last one being space, I'm doing my best to avoid using to much space but still provide a quick enough implementation so that models gets cached preventing multiple computations of the same models to avoid freezing but this comes at the cost of more memory which can be a sparce resource when it come to java game...
- For the layered panels I will address theses issues for the next updates too  shouldn't be too hard of a fix but thanks for noticing that (it's mostly due to the fact that I don't have any proper testing procedure but well I still try my best to avoid theses issues xD)
- I Hear your complains about the way certain block placements are done. I do agree that the placement can be a bit tedious and I think it need a remodel or some visual feedback to it. In a mod I started making years ago, I added a small visualizer on the block to see which part was being targeted but i found that it looked a bit weird and that it didn't really match the theme of the game so I might just add a ghost visualization of the block before placing it (I'm not there yet though I'd like to get some more shapes added to the mod and theses can take quite some time :-( )\
Here is an example of how it looked (I inspired myself from tweakeroo back then):\
![cursor_indicator](https://variouscolors.net/game/wp-content/uploads/2020/09/tweakeroo-direction.webp)\
Currently for the stairs it would look more like just a cross going from the corners and for the small_cube a cross going for centers of the edges so for the slabs I could use the one from the image so that placing on other sides doesn't only depend on the clicked side.
- Finally now that you're here, I'd like to get your opinion on making the wall work in all axis/direction would be a cool thing to have or would it just make things more confusing. I need to know that because making it works on multiple axis is quite easy when it come to logic bit might be a bit too overboard as the total amount of states would be 3750 against an incredible 324 for the normal walls. (Also I need to remodel it because otherwise transparent walls will have textures inside which are weird)\
Btw making it works in all axis means it would also work the same way on a ceiling as it works on the floor and same for walls but if nothing in between the wall would look like this:\
![image](https://github.com/DriHut/ReFramed/assets/63098227/73cbf4a9-d666-4968-8aed-0c668c5e5f12)
\
\
\
Here you go! Took me a bit of text answering you feedback but everything should be here ^^.

**Comment by UnintelligentIdiot** [comment]
*You don't need to worry about the other mods and how they will be implemented, this mod as how I find it is expansion of a single 16x16x16 cube or a vanilla mc block not moded ones. And as far as I understand this mod without any coding knowledge, the block should use the block models and divides them in how ever shape the block is, and the screwdriver uses the block states if there is any, so I believe its ok to add more to it. And its ok to not work with some mod blocks, if they are not 16x16x16 obviously.
*No worries if you need someone to test stuff I really like to do that kind of stuff.
*About the block placements without adding new stuff you can just make the blocks separate it would be more easy for placing, less confusion (because it took me too long to figure out how to place upper slab xD) and less code for one block.
*From a builder perspective I need you to do the walls fully <3
What I mean is when you are going to add the key you will need to do the walls completely textured I don't know how much there are missing side textures, but there are a lot missing textures in vanilla which I have to use "Krista's Better Walls" texture pack. It would be cool to have vertical walls as well, but please make them separate blocks as it really is hard to place stuff.. the crafting could be just 1=1.

What blocks do you plan on adding next?

**Comment by UnintelligentIdiot** [comment]
@DriHut you have ghosted us ha? 😭👍

**Comment by DriHut** [comment]
No sorry I'm quite busy lately. I won't be working on the project in the next two weeks for sure. Though I did read you message and will provide improvements for the slab by splitting it into two blocks and the walls are planned to be fully made (I knew the issue of the missing textures with the debug stick) And I will now rely more on your testing capabilities. (BTW less code for one block isn't necessarily true for the slabs it will be more)

**Comment by UnintelligentIdiot** [comment]
@DriHut Hey!! Sorry for not responding earlier I was in the hospital.
I had a furniture mod for forge and I remember fraction of it, basically I used Gecko-Lib mod and just rewriting already existing blocks like glazed terracotta or something that has the block-state that I need and it worked with the forge version of the Framed Blocks. I had like a half chair and a random block on the other side it was funny but I never really thought anything of it. I will try to recreate the mod and send you a screenshot, but don't have high hopes because my coding is really bad.
From what I understand your mod is using the texture of the block not the model, hence there are so many bugs with the other mods.

**Comment by DriHut** [comment]
Hey yes there are compatibility issues with other mods when it comes to rendering related mods. 
If there are things you want to add to the mod and I don't plan on adding them you can create a fork and a pull request into this project so on my side I would just need to review the code and if needed do some changes. And thoses things would still be in there thanks to you like I did for the backport to 1.20.1.
The mod works by extracting the textures from the model and then applies theses textures to the frame models.
Anyway I hope You're doing fine after being in the hospital.

**Comment by UnintelligentIdiot** [comment]
@DriHut , I'm doing much better thank you.
But I'm saying that you shouldn't extract textures. You should extract the block model, this is what I have been saying all the time, the code in its self is alright but its completely in the wrong direction. Because some block model have multiple .png's or they are not made by the same format,(like 16x16x16) so you need to extract the block model(block.json) because there is done how it should be portrayed in the game.

**Comment by DriHut** [comment]
You cannot change the shape of a model you can just apply textures to existing shape and that works fine since I keep all the model textures. so to put it sequentially I get the model for a given state from the renderer and apply this model textures to a frame model which I'm pretty sure is the only way to do it.

**Comment by UnintelligentIdiot** [comment]
No. You can extract the block model and cut it in the shape that you want. It is much easier then getting the model then getting the texture and to apply it
You are essentially skiping a step

**Comment by DriHut** [comment]
do you have an example cuz I can't really see how to do that. I believe that it might be possible I just don't see how ^^.

**Comment by UnintelligentIdiot** [comment]
I don't my laptop got destroyed in the accident. But you are already taking the block model while trying to find the texture u have already figured it out.
